packages/_shared/src/FieldConnector.ts

```diff
@@ -122,6 +123,20 @@ export class FieldConnector<ValueType> extends React.Component<
     });
   }
 
+  componentDidUpdate(): void {
+    const { field } = this.props;
+
+    /**
+     * Only propagate the needed change when the value really changed
+     * so we don't end up in an infinite state loop
+     */
+    this.unsubscribeDisabled = field.onIsDisabledChanged((disabled: boolean) => {
+      if (this.state.disabled !== disabled) {
+        this.setState({ disabled });
+      }
+    });
+  }
```

Comment:
Why not write it like this already in `componentDidMount` so we always check for `disabled` beforehand?

Reply:
I am not sure what you exactly mean. `componentDidMount` is just called once in the lifecycle. So any following updates of the permissions are not propagated.